src/backend/mpi/swdsm.cpp

```diff
@@ -1094,35 +1210,245 @@ void storepageDIFF(std::size_t index, std::uintptr_t addr){
+			sharer_num_locks += mpi_lock_sharer[i][j].get_num_locks();
+
+			/* Get local lock stats */
+			sharer_local_lock_time 		+= 	mpi_lock_sharer[i][j].get_local_lock_time();
+			sharer_local_hold_time 		+= 	mpi_lock_sharer[i][j].get_local_hold_time();
+			if(mpi_lock_sharer[i][j].get_max_local_lock_time() > sharer_local_max_lock_time){
+				sharer_local_max_lock_time = mpi_lock_sharer[i][j].get_max_local_lock_time();
+			}
+			if(mpi_lock_sharer[i][j].get_max_local_hold_time() > sharer_local_max_hold_time){
+				sharer_local_max_hold_time = mpi_lock_sharer[i][j].get_max_local_hold_time();
+			}
+
+			/* Get mpi lock stats */
+			sharer_mpi_lock_time 			+= 	mpi_lock_sharer[i][j].get_mpi_lock_time();
+			sharer_mpi_unlock_time 		+= 	mpi_lock_sharer[i][j].get_mpi_unlock_time();
+			sharer_mpi_hold_time 			+=	mpi_lock_sharer[i][j].get_mpi_hold_time();
+			if(mpi_lock_sharer[i][j].get_max_mpi_lock_time() > sharer_mpi_max_lock_time){
+				sharer_mpi_max_lock_time = mpi_lock_sharer[i][j].get_max_mpi_lock_time();
+			}
+			if(mpi_lock_sharer[i][j].get_max_mpi_unlock_time() > sharer_mpi_max_unlock_time){
+				sharer_mpi_max_unlock_time = mpi_lock_sharer[i][j].get_max_mpi_unlock_time();
+			}
+			if(mpi_lock_sharer[i][j].get_max_mpi_hold_time() > sharer_mpi_max_hold_time){
+				sharer_mpi_max_hold_time = mpi_lock_sharer[i][j].get_max_mpi_hold_time();
+			}
+		}
+	}
+	/** Get averages */
+	sharer_local_avg_lock_time 	= sharer_local_lock_time / sharer_num_locks;
+	sharer_local_avg_hold_time 	= sharer_local_hold_time / sharer_num_locks;
+	sharer_mpi_avg_lock_time 		= sharer_mpi_lock_time / sharer_num_locks;
+	sharer_mpi_avg_unlock_time 	= sharer_mpi_unlock_time / sharer_num_locks;
+	sharer_mpi_avg_hold_time 		= sharer_mpi_hold_time / sharer_num_locks;
+
+
+	/** Nicely format and print the results */
+	MPI_Barrier(MPI_COMM_WORLD);
+	if(workrank == 0){
+		/** Adjust memory size */
+		double mem_size_readable = size_of_all;
+		std::vector<const char*> sizes = { "B ", "KB", "MB", "GB", "TB" };
+		std::size_t order = 0;
+		while (mem_size_readable >= 1024 && order < sizes.size()-1) {
+			order++;
+			mem_size_readable /= 1024;
+		}
+
+		/* Print general information */
+		printf("\n#################################" YEL" ArgoDSM statistics " RESET "##################################\n");
+		printf("#  memory size: %12.2f%s  page size (p): %10dB   cache size: %13ldp\n",
+				mem_size_readable, sizes[order], pagesize, cachesize);
+		printf("#  write buffer size: %6ldp   write back size: %8ldp   CACHELINE: %14ldp\n",
+				env::write_buffer_size()/CACHELINE,
+				env::write_buffer_write_back_size()/CACHELINE,
+				CACHELINE);
+		printf("#  allocation policy: %6ld    policy block size: %6ldp   load size: %14ldp\n",
+				env::allocation_policy(), env::allocation_block_size(), env::load_size());
+		printf("#  active time: %12.4fs   init time: %14.4fs\n",
+				stats.exectime, stats.inittime);
+		printf("\n");
+	}
+
+	/* Print node information */
+	if(print_level > 1) {
```

Comment:
Isn't this check redundant and should be removed so that the code is not indented one level deeper?
(There is a `if (print_level == 0) return` at the beginning of this function.)

Perhaps move this check in its place instead and have the whole function body inside it instead of having the function as a noop when print_level == 0.

Reply:
I think it is needed as there are technically 4 print levels that are inclusive of all lower levels.
0 - Disable printing alltogether
1 - Print only general system information
2 - Print basic node statistics
3 - Print detailed node statistics

I opted to return for `print_level == 0` instead of checking for `print_level > 0` and causing additional indentation. The check you are referring to checks for `print_level` 2 or higher.